crates/linker/src/lib.rs

```diff
@@ -123,15 +123,87 @@ fn make_dummy_dll_symbols(
     custom_names
 }
 
-fn generate_dynamic_lib(target: &Triple, custom_names: &[String], dummy_lib_path: &Path) {
-    if !dummy_lib_is_up_to_date(target, dummy_lib_path, custom_names) {
-        let bytes = crate::generate_dylib::generate(target, custom_names)
+fn generate_dynamic_lib(target: &Triple, dummy_dll_symbols: &[String], dummy_lib_path: &Path) {
+    if !dummy_lib_is_up_to_date(target, dummy_lib_path, dummy_dll_symbols) {
+        let bytes = crate::generate_dylib::generate(target, dummy_dll_symbols)
             .unwrap_or_else(|e| internal_error!("{e}"));
 
-        std::fs::write(dummy_lib_path, &bytes).unwrap_or_else(|e| internal_error!("{e}"))
+        std::fs::write(dummy_lib_path, &bytes).unwrap_or_else(|e| internal_error!("{e}"));
+
+        if let target_lexicon::OperatingSystem::Windows = target.operating_system {
+            generate_import_library(dummy_lib_path, dummy_dll_symbols);
+        }
+    }
+}
+
+fn generate_import_library(dummy_lib_path: &Path, custom_names: &[String]) {
+    let def_file_content = generate_def_file(custom_names).expect("write to string never fails");
+
+    let mut def_path = dummy_lib_path.to_owned();
+    def_path.set_extension("def");
+
+    std::fs::write(def_path, def_file_content.as_bytes())
+        .unwrap_or_else(|e| internal_error!("{e}"));
+
+    let mut def_filename = PathBuf::from(generate_dylib::APP_DLL);
+    def_filename.set_extension("def");
+
+    let mut lib_filename = PathBuf::from(generate_dylib::APP_DLL);
+    lib_filename.set_extension("lib");
+
+    let zig = std::env::var("ROC_ZIG").unwrap_or_else(|_| "zig".into());
+
+    // use zig to generate the .lib file. Here is a good description of what is in an import library
+    //
+    // > https://www.codeproject.com/Articles/1253835/The-Structure-of-import-Library-File-lib
+    //
+    // For when we want to do this in-memory in the future. We can also consider using
+    //
+    // > https://github.com/messense/implib-rs
```

Comment:
Interesting - so this means the Windows surgical linker relies on having Zig installed for now? (But maybe we could remove that dependency using `implib-rs`?)